internal/controllers/machineset/machineset_controller.go

```diff
@@ -969,12 +994,47 @@ func (r *Reconciler) getMachineNode(ctx context.Context, cluster *clusterv1.Clus
 
 func (r *Reconciler) reconcileUnhealthyMachines(ctx context.Context, cluster *clusterv1.Cluster, ms *clusterv1.MachineSet, filteredMachines []*clusterv1.Machine) (ctrl.Result, error) {
 	log := ctrl.LoggerFrom(ctx)
+
+	// Calculate how many in flight machines we should remediate.
+	// By default, we allow all machines to be remediated at the same time.
+	maxInFlight := len(filteredMachines)
+
+	// If the MachineSet is part of a MachineDeployment, only allow remediations if
+	// it's the desired revision.
+	if util.HasOwner(ms.GetOwnerReferences(), clusterv1.GroupVersion.String(), []string{"MachineDeployment"}) {
+		owner, err := r.getOwnerMachineDeployment(ctx, ms)
+		if err != nil {
+			return ctrl.Result{}, err
+		}
+
+		if owner.Annotations != nil && ms.Annotations != nil && owner.Annotations[clusterv1.RevisionAnnotation] != ms.Annotations[clusterv1.RevisionAnnotation] {
+			// MachineSet is part of a MachineDeployment but isn't the current revision, no remediations allowed.
+			return ctrl.Result{}, nil
+		}
+
+		if owner.Spec.Strategy != nil && owner.Spec.Strategy.Remediation != nil {
+			if owner.Spec.Strategy.Remediation.MaxInFlight != nil {
+				var err error
+				replicas := int(ptr.Deref(owner.Spec.Replicas, 1))
+				maxInFlight, err = intstr.GetScaledValueFromIntOrPercent(owner.Spec.Strategy.Remediation.MaxInFlight, replicas, true)
+				if err != nil {
+					return ctrl.Result{}, fmt.Errorf("failed to calculate maxInFlight to remediate machines: %v", err)
+				}
+				log.Info("Remediation strategy is set", "maxInFlight", maxInFlight, "replicas", replicas)
```

Comment:
log level 0 seems pretty verbose. We'll hit this on every reconcile and it seems more like a debug message

Reply:
I think it would be good to drop this log message entirely and just add k/v pairs to the logger. So that the log messages below contain the information